src/main/java/minicraft/level/Level.java

```diff
@@ -532,7 +567,7 @@ public void setTile(int x, int y, Tile t, int dataVal) {
 
 	public int getData(int x, int y) {
 		if (x < 0 || y < 0 || x >= w || y >= h) return 0;
-		return data[x + y * w] & 0xff;
```

Comment:
Are you sure this doesn't break anything?

Reply:
Yes, as the returning type is `int`, if it is limited to the range of `byte`, the possible data is really limited, it is still limited to the range of `short`.

Reply:
Maybe it is limited for a reason?

Reply:
It was byte.

Reply:
There was no a tile using data for larger than their max health.

Reply:
The resting bits might be used somewhere else.

Reply:
No, I have checked. Also, it also does not make sense that `setData` accepts data in range of `short`, but `getData` strip data in range of `byte`. Before the change from `byte` to `short`, there are *a lot of* code using `& 0xFF`, which are the places that I think are weird.

Reply:
If ain't broke, don't fix it.

Reply:
When I saw a line `/// The "& 0xffff" is a common way to convert a short to an unsigned int, which basically prevents negative values... except... this doesn't do anything if you flip it back to a short again...` and after a slight investigation, I also find that there would be problem in fertilization when the value exceeds 32767. The fix should be `& 0xFFFF` instead. By the way, I have also noticed that there is `if (id < 0) id += 32768;` in TIles, but `(int)((short) 32768)` is `-32768`.